Shared/Repository/GenericRepository.swift

```diff
@@ -0,0 +1,16 @@
+//
+//  GenericRepository.swift
+//  InternxtDesktop
+//
+//  Created by Patricio Tovar on 23/7/24.
+//
+
+import Foundation
+
+protocol GenericRepositoryProtocol {
+    associatedtype T
+    
+    func addSyncedNode(_ entity: T) throws
+    func find(url: URL, deviceId: Int) -> T?
+    func editSyncedNodeDate(remoteUuid: String, date: Date) throws
```

Comment:
You can implement a `SyncedNodesRepository` that implements the `GenericRepositoryProtocol`, but functions such `editSyncedNodeDate` should be owned by the `SyncedNodeRepository` iitself, rahter than the protocol.

Imagine we create in the future a new repository called `CachedFilesRepository` and it extends the `GenericRepositoryProtocol`, it will ask us to implement the `addSyncedNode` and `editSyncedNodeDate` members, but those makes no sense for the `CachedFilesRepository`